pipeline_dp/private_contribution_bounds.py

```diff
@@ -0,0 +1,127 @@
+from dataclasses import dataclass
+from functools import lru_cache
+from typing import List
+
+import pipeline_dp
+from pipeline_dp.dp_computations import ScalarNoiseParams, compute_dp_count_noise_std, ExponentialMechanism
+
+from pipeline_dp.histograms import Histogram
+
+
+class PrivateL0Calculator:
+    """Calculates differentially-private l0 bound (i.e. max_partitions_contributed)."""
+
+    def __init__(self,
+                 params: pipeline_dp.CalculatePrivateContributionBoundsParams,
+                 partitions, histograms, backend) -> None:
+        self._params = params
+        self._backend = backend
+        self._partitions = partitions
+        self._histograms = histograms
+
+    @dataclass
+    class Inputs:
+        l0_histogram: Histogram
+        number_of_partitions: int
+        possible_contribution_bounds: List[int]
+
+    @lru_cache(maxsize=None)
```

Comment:
This would cache calculation per calculator, right? Do we need to call it more than once?

Reply:
sg, it makes sense.